org.OpenT2T.Sample.SuperPopular.Helpers/js/index.js

```diff
@@ -0,0 +1,169 @@
+'use strict';
+
+//var request = require('request');
+var https = require('https');
+var q = require('q');
+
+//this is our base, we each call has its own set of options
+ var options = 
+ {
+        protocol: 'https:',
+        host: 'api.wink.com'     
+ };
+    
+var  accessToken;  //we will need this when creating the headers
+      
+module.exports = 
+{
+    //get the initial parameters to build our target endpoint
+   initWinkApi: function(apiEndpoint, deviceID, token)
+   {
+       options.path = '/' + apiEndpoint + '/' + deviceID;
+       accessToken = token; //the specific token
+        
+   },
+ 
+   sendDesiredStateCommand: function(apiField,value,callback) 
```

Comment:
There's no need for all of these methods to take a callback. If a caller wants to do something after the async operation they can call .then() on the returned promise.